scripts/tests/cross-test-ci-all.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+set -x
+
+# Test current versions against gatewayd v0.2
+export FM_GATEWAYD_BASE_EXECUTABLE="nix run git+https://github.com/fedimint/fedimint.git?ref=releases/v0.2#gatewayd --"
+# shellcheck disable=SC2260
+$FM_GATEWAYD_BASE_EXECUTABLE --help || true
+./scripts/tests/test-ci-all.sh
```

Comment:
You mentioned in the PR description this script is just an example, but worth highlighting if we use `test-ci-all.sh` in a backwards compatibility run the backend tests will still use clients/servers on the same version.